pkg/envoy/proxy.go

```diff
@@ -114,6 +119,13 @@ func (e *envoy) args(fname string, bootstrapConfig string) []string {
 	proxyLocalAddressType := "v4"
 	if network.AllIPv6(e.NodeIPs) {
 		proxyLocalAddressType = "v6"
+	} else if DualStackEnv {
+		// If dual-stack, it may be [IPv4, IPv6] or [IPv6, IPv4]
+		// So let the first ip family policy to decide its DNSLookupFamilyIP policy
+		ipFamily := network.CheckIPFamilyTypeForFirstIPs(e.NodeIPs)
```

Comment:
Yes, it does

Reply:
Hi @howardjohn, basically, I can remove this change to follow the original logic, it means that the `proxyLocalAddressType` can be `v4` even it's Dual Stack. BTW, `v4` and `v6` are the only 2 values for  `proxyLocalAddressType`.

Reply:
Please explain *in the code* why its done.

Reply:
Hi @howardjohn, in fact, `proxyLocalAddressType` does not impact the issue, I just found that the args showing as below `--local-address-ip-version v4` even if the IPv6 is the first in Dual Stack:

```
Envoy command: [-c etc/istio/proxy/envoy-rev.json --drain-time-s 45 --drain-strategy immediate --local-address-ip-version v4 --file-flush-interval-msec 1000 --disable-hot-restart --allow-unknown-static-fields --log-format %Y-%m-%dT%T.%fZ    %l      envoy %n %g:%#  %v      thread=%t -l warning --component-log-level misc:error --concurrency 2]
```
So want to make it in the same format as before `DNSLookupFamily` settings. And I removed this logic to follow the original logic. It can be working now.